### PR TITLE
Implemented client key injection into PaymentMethodsConfiguration, an…

### PR DESCRIPTION
**Pull request by mohammedDehairy**
## Summary
Implemented client key injection into PaymentMethodsConfiguration, and propagating the client key to DropInComponent and all the paymentMethod components, also added unit tests.

## Tested scenarios
1. Tested `ComponentManager` injecting the client key into its created components.
2. Tested MBWay payment method decoding from `paymentMethods/` endpoint response.
3. Tested `ComponentManager` scrapping `MBWayComponent` in case client key is not configured.

### CODE REVIEW

File: AdyenDropIn/DropInActionComponent.swift [hunk 1]
```diff
@@ -100,6 +100,11 @@ public final class DropInActionComponent: ActionComponent {
     }
     
     private func perform(_ action: AwaitAction) {
+        guard environment.clientKey != nil else {
+            // swiftlint:disable:next line_length
+            assertionFailure("Failed to instantiate AwaitComponent because client key is not configured, Please supply the client key in the PaymentMethodsConfiguration if using DropInComponent, or DropInActionComponent.environment.clientKey if using DropInActionComponent separately.")
```
Comment (descorp) [inline AdyenDropIn/DropInActionComponent.swift:105]: `configured, Please supply ` -> `configured. Please supply `

Comment (descorp) [inline AdyenDropIn/DropInActionComponent.swift:105]: `DropInActionComponent.environment.clientKey if using DropInActionComponent separately`

Maybe we can move `clientKey` as calculated property of DropInActionComponent? For convenience.
`environment.clientKey` is a bit more "internal" implementation detail..

File: AdyenDropIn/Utilities/ComponentManager.swift [hunk 2]
```diff
@@ -130,7 +132,12 @@ internal final class ComponentManager {
         return component
     }
     
-    private func createMBWayComponent(_ paymentMethod: MBWayPaymentMethod) -> MBWayComponent {
+    private func createMBWayComponent(_ paymentMethod: MBWayPaymentMethod) -> MBWayComponent? {
+        guard configuration.clientKey != nil else {
+            // swiftlint:disable:next line_length
+            adyenPrint("Failed to instantiate MBWayComponent because client key is not configured, Please supply the client key in the PaymentMethodsConfiguration.")
```
Comment (descorp) [inline AdyenDropIn/Utilities/ComponentManager.swift:138]: `configured, Please` -> `configured. Please`

Comment (descorp) [inline AdyenDropIn/Utilities/ComponentManager.swift:138]: Should we mention DrpoInACtionComponent here, too?

Comment (mohammedDehairy) [reply AdyenDropIn/Utilities/ComponentManager.swift:138]: here its means that they are using dropInComponent, and `DropInComponent` is instantiating the `DrpoInACtionComponent` and making sure the client key is supplied to it.

Comment (mohammedDehairy) [reply AdyenDropIn/Utilities/ComponentManager.swift]: done

File: AdyenDropIn/DropInActionComponent.swift (hunk 1 above)
Comment (mohammedDehairy) [reply AdyenDropIn/DropInActionComponent.swift]: 👍 good idea

Comment (mohammedDehairy) [reply AdyenDropIn/DropInActionComponent.swift]: 👍

File: Adyen/Components/Base/Component.swift [hunk 3]
```diff
@@ -29,6 +33,17 @@ public extension Component {
         }
     }
     
+    /// :nodoc:
```
Comment (descorp) [inline Adyen/Components/Base/Component.swift:36]: We can actually add here some user-friendly description. This is something that component implementers could possibly use..

File: Adyen/Components/Base/Component.swift [hunk 4]
```diff
@@ -1,5 +1,5 @@
 //
-// Copyright (c) 2019 Adyen N.V.
+// Copyright (c) 2020 Adyen N.V.
```
Comment (descorp) [inline Adyen/Components/Base/Component.swift:2]: This could be reverted 😅

File: Adyen/Components/Base/Component.swift (hunk 3 above)
Comment (mohammedDehairy) [reply Adyen/Components/Base/Component.swift:36]: there is documentation in the `Component` protocol for the `clientKey` var, right? we have kind of a convention that if a class or struct inherits from/implements a class/protocol, the overriding declaration/implementation should show :nodoc:, as the parent class/protocol has the documentation, so this falls under this rule/convention I guess.

File: Adyen/Components/Base/Component.swift (hunk 4 above)
Comment (mohammedDehairy) [reply Adyen/Components/Base/Component.swift]: reverted it.

**Pull request merged**